### ISSUE TITLE
Error message after aborting file transfer

### ISSUE THREAD

**Issue by zensubz** [masneyb/gftp#125]
I have come across the following error message while I was running gftp-gtk (via terminal), and this came-up after I had aborted a file-transfer (from remote server). I am using the latest git commit of gftp (fffed76). Just sharing it here, just in case:

```** (gftp-gtk:1900): CRITICAL **: 16:04:57.850: rfc959_abort_transfer: assertion 'request->datafd > 0' failed```

**Comment by zensubz** [comment]
<b>Update:</b> I could restart (resume) file-transfer, and downloaded the file successfully from the remote server. The above error msg didn't possibly affect anything related to the integrity of the file actually downloaded, perhaps, it was just a warning msg. Closing the issue.

**Comment by wdlkmpx** [comment]
Yeah as long as it doesn't segfault and no further issues are encountered, everything is ok I guess. C code is very delicate, very fragile. Nobody can change it. This videos explains everything in detail
https://www.youtube.com/watch?v=Vor6Yul7CMg

I used to upload iso files in 2019 with gftp, but my connectivity with the server was very bad, and gFTP apparently can't handle shaky connections very well, I needed to limit the upload speed. But now that I my upload speed is about 10MB/s, I just use the browser to upload anything.

So I decided to use gftp again to upload random stuff to SourceForge, aaaaaand...  after playing with the remote directory tree for some minutes... I got a crash.

I also made gftp crash days ago with ftp.bit.nl, the internet connection was already gone, but the network tray was still showing connection with the router I guess?, and so apps thought the connection is up, after several chdirs and no visible changes in the remote directory tree, gftp crashed. This particular crash is hard to reproduce.

gFTP should always have a thread to verify the connection with the server, once a connection is made, many widgets should become disabled until the connection ends, (of course this should not apply to UDP connections). I have ideas, but I don't think I'm good enough to implement them.
